### PR TITLE
Move Utf8String and Utf8Span code to src/libraries.

### PR DESCRIPTION
**Pull request by eerhardt**
This is the first step in getting Utf8String and Utf8Span working on netstandard2.0.

Contributes to #29442

Should also help contribute to #32481.

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems [hunk 1]
```diff
@@ -1808,4 +1808,23 @@
   <ItemGroup Condition="'$(FeaturePortableThreadPool)' == 'true' or '$(FeatureThreadInt64PersistentCounter)' == 'true'">
     <Compile Include="$(MSBuildThisFileDirectory)System\Threading\ThreadInt64PersistentCounter.cs" />
   </ItemGroup>
+  <ItemGroup Condition="'$(FeatureUtf8String)' == 'true'">
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems]: Presumably this is not currently set when building mono?

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems (hunk 1 above)
Comment (eerhardt) [reply src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems]: No - it is only defined here (from searching the whole repo):

https://github.com/dotnet/runtime/blob/a80e6d4c69fcfa1e17e17130c144a567b23a12a9/src/coreclr/clr.featuredefines.props#L45

Comment (eerhardt) [reply src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems]: But this work should help when we want to enable Utf8String on mono (#32481).

**Review by GrabYourPitchforks — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems (hunk 1 above)
Comment (EgorBo) [reply src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems]: As far as I understand Utr8String doesn't require runtime support so it should work in mono as is (once this variable is defined)

**Pull request merged**